src/transformers/models/clip/modeling_tf_clip.py

```diff
@@ -1375,4 +1410,42 @@ def call(
         return outputs
 
     def serving_output(self, output: TFCLIPOutput) -> TFCLIPOutput:
-        return output
+        text_hs = (
+            tf.convert_to_tensor(output.text_model_output.hidden_states) if self.config.output_hidden_states else None
+        )
+        text_attns = (
+            tf.convert_to_tensor(output.text_model_output.attentions) if self.config.output_attentions else None
+        )
+        text_model_output = TFBaseModelOutputWithPooling(
+            last_hidden_state=output.text_model_output.last_hidden_state,
+            pooler_output=output.text_model_output.pooler_output,
+            hidden_states=text_hs,
+            attentions=text_attns,
+        )
+
+        vision_hs = (
+            tf.convert_to_tensor(output.vision_model_output.hidden_states)
+            if self.config.output_hidden_states
+            else None
+        )
+        vision_attns = (
+            tf.convert_to_tensor(output.vision_model_output.attentions) if self.config.output_attentions else None
+        )
+        vision_model_output = TFBaseModelOutputWithPooling(
+            last_hidden_state=output.vision_model_output.last_hidden_state,
+            pooler_output=output.vision_model_output.pooler_output,
+            hidden_states=vision_hs,
+            attentions=vision_attns,
+        )
+
+        output = TFCLIPOutput(
+            loss=output.loss,
+            logits_per_image=output.logits_per_image,
+            logits_per_text=output.logits_per_text,
+            text_embeds=output.text_embeds,
+            image_embeds=output.image_embeds,
+            text_model_output=text_model_output,
+            vision_model_output=vision_model_output,
+        )
+
+        return output.to_tuple()
```

Comment:
Converting to a tuple would break the assumption in the type hint. Can we remove this line?

Reply:
Updated, but unfortunately the test fails because `TFCLIPOutput` is a nested dataclass---I'm not 100% if TF can trace through nested dataclasses or not, or if the reason it's failing lies somewhere else